Source/Core/AudioCommon/Mixer.cpp

```diff
@@ -49,128 +74,154 @@ unsigned int CMixer::MixerFifo::Mix(short* samples, unsigned int numSamples, boo
 	{
 		aid_sample_rate = aid_sample_rate * (framelimit - 1) * 5 / VideoInterface::TargetRefreshRate;
 	}
-
-	const u32 ratio = (u32)(65536.0f * aid_sample_rate / (float)m_mixer->m_sampleRate);
-
-	s32 lvolume = m_LVolume.load();
-	s32 rvolume = m_RVolume.load();
-
-	// TODO: consider a higher-quality resampling algorithm.
-	for (; currentSample < numSamples * 2 && ((indexW-indexR) & INDEX_MASK) > 2; currentSample += 2)
+	// ratio = 1 / upscale_factor = stepsize for each sample
+	// e.g. going from 32khz to 48khz is 1 / (3 / 2) = 2 / 3
+	// note because of syncing and framelimit, ratio will rarely be exactly 2 / 3
+	float ratio = aid_sample_rate / (float)m_mixer->m_sample_rate;
+	float l_volume = (float)m_lvolume.load() / 256.f;
+	float r_volume = (float)m_rvolume.load() / 256.f;
+	// for each output sample pair (left and right),
+	// linear interpolate between current and next sample
+	// increment output sample position
+	// increment input sample position by ratio, store fraction
+	// QUESTION: do we need to check for NUM_CROSSINGS samples before we interpolate?
+	// seems to work fine as is
+	for (; current_sample < numSamples * 2 && ((write_index - read_index) & INDEX_MASK) > 0; current_sample += 2)
```

Comment:
The old loop did check for more than 2 samples in the queue, now this is out-of-bound access.
Maybe a virtual interface to query for the required window size?

Hint: Cubic interpolation accesses the sample at "left_input_index + 7"